### PR TITLE
dns: add dns.rrtype keyword v6

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Feature #6666

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6666) ticket: https://redmine.openinfosecfoundation.org/issues/6666

Previous PR: https:https://github.com/OISF/suricata/pull/10435

Describe changes:
- rebased with master
- renamed the unit test for rrtype in `detect.rs` file.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1655

### CODE REVIEW

File: rust/src/dns/detect.rs [hunk 1]
```diff
@@ -50,6 +50,39 @@ pub extern "C" fn rs_dns_opcode_match(
...
+    if flags & Direction::ToServer as u8 != 0 {
+        if let Some(request) = &tx.request {
+            for i in 0..request.queries.len() {
+                if detect_match_uint(detect, request.queries[i].rrtype) {
+                    return 1;
+                }
+            }
+        } else {
+            return 0;
+        }
```
Comment (inashivb) [inline rust/src/dns/detect.rs:68]: unnecessary else block?

File: rust/src/dns/detect.rs [hunk 2]
```diff
@@ -50,6 +50,39 @@ pub extern "C" fn rs_dns_opcode_match(
...
+    } else {
+        // Not to server or to client??
+        return 0;
+    };
```
Comment (inashivb) [inline rust/src/dns/detect.rs:82]: unnecessary else blocks?

**Review by inashivb — CHANGES_REQUESTED** [review]
Looking pretty good. :)
Just some nits inline.

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10481

**Pull request closed without merging**